### PR TITLE
Maneuverreviewchanges

### PR DESCRIPTION
**Pull request by kdemski**
new maneuver class
lower limits on altitude for initiating maneuvers
reading from resources once on app startup
other updates

### CODE REVIEW

**Comment by kdemski** [comment]
move isManuverRunning to either main or FlightManuver

**Comment by kdemski** [comment]
move the two functions from loader to FlightManeuvers, move applyVsq into utils and make it take a list as a parameter

**Comment by kdemski** [comment]
make another xml file for the maneuver arrays

File: app/src/main/java/com/aeriosense/projecttalon/missionPlanner/FlightManeuvers.java [hunk 1]
```diff
@@ -14,6 +16,8 @@
 
 public class FlightManeuvers extends ContextWrapper  {
     private static ArrayList<List> maneuverVsqs = new ArrayList<>();
+    public static FlightPlanData maneuverData = new FlightPlanData();
+    public static boolean maneuverRunning = false;
```
Comment (owen927) [inline app/src/main/java/com/aeriosense/projecttalon/missionPlanner/FlightManeuvers.java]: these two should not be static

File: app/src/main/java/com/aeriosense/projecttalon/missionPlanner/FlightManeuvers.java [hunk 2]
```diff
@@ -39,6 +43,37 @@ public void initManeuvers(){
...
+            msTemp.add(maneuverNameList[i]);
+        }
+
+        int index;
+        if(msTemp.contains(maneuverName)){
+            index = msTemp.indexOf(maneuverName);
+            loaded = true;
+            List<MissionVector> vsq = FlightManeuvers.getManeuverVsq(index);
+            maneuverData.clear();
+            VectorSequenceTools.applyVsq(maneuverData, MissionLoader.towerReferenceLon, MissionLoader.towerReferenceLat, MissionLoader.towerReferenceAlt, MissionLoader.towerReferenceDirection, vsq);
```
Comment (owen927) [inline app/src/main/java/com/aeriosense/projecttalon/missionPlanner/FlightManeuvers.java]: should do mauverData =   VectorSequenceTools.applyVsq( MissionLoader.towerReferenceLon, MissionLoader.towerReferenceLat, MissionLoader.towerReferenceAlt, MissionLoader.towerReferenceDirection, vsq);
you would not grab the tower reference from missionloader.  they exist in main activity as     private double droneLocationLat = 181, droneLocationLng = 181;
    private double droneHeading = 0;
    private float droneAlt = 999;

File: app/src/main/java/com/aeriosense/projecttalon/missionLoader/MissionLoader.java [hunk 3]
```diff
@@ -24,11 +23,11 @@
...
-    public double towerLocationLat = 0, towerLocationLon = 0;
-    public double towerReferenceLat = 0, towerReferenceLon = 0;
-    public double towerReferenceDirection = 0;
-    public float towerReferenceAlt = 0;
-    public static boolean maneuverRunning = false;
+    public static double towerLocationLat = 0, towerLocationLon = 0;
+    public static double towerReferenceLat = 0, towerReferenceLon = 0;
+    public static double towerReferenceDirection = 0;
+    public static float towerReferenceAlt = 0;
+
```
Comment (owen927) [inline app/src/main/java/com/aeriosense/projecttalon/missionLoader/MissionLoader.java]: there should not be static either

File: app/src/main/java/com/aeriosense/projecttalon/missionPlanner/FlightManeuvers.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import com.aeriosense.projecttalon.missionLoader.FlightPlanData;
+import com.aeriosense.projecttalon.missionLoader.MissionLoader;
+import com.aeriosense.projecttalon.missionLoader.MissionVector;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class FlightManeuvers extends ContextWrapper  {
+    private static ArrayList<List> maneuverVsqs = new ArrayList<>();
```
Comment (owen927) [inline app/src/main/java/com/aeriosense/projecttalon/missionPlanner/FlightManeuvers.java]: should this be static?

File: app/src/main/java/com/aeriosense/projecttalon/missionPlanner/FlightManeuvers.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    public FlightPlanData maneuverData = new FlightPlanData();
```
Comment (owen927) [inline app/src/main/java/com/aeriosense/projecttalon/missionPlanner/FlightManeuvers.java]: this should be private

File: app/src/main/java/com/aeriosense/projecttalon/missionPlanner/FlightManeuvers.java (hunk 4 above)
Comment (kdemski) [inline app/src/main/java/com/aeriosense/projecttalon/missionPlanner/FlightManeuvers.java]: edited to non-static

File: app/src/main/java/com/aeriosense/projecttalon/missionPlanner/FlightManeuvers.java (hunk 5 above)
Comment (kdemski) [inline app/src/main/java/com/aeriosense/projecttalon/missionPlanner/FlightManeuvers.java]: edited to private

**Pull request merged**